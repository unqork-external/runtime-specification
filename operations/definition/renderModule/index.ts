export * from './renderModule.model'
export * from './renderModule.options'
