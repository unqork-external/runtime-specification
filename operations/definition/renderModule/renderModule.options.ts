import { Description, Required } from '@tsed/schema'

import { TargetedOperationOptions } from '../../interface'

export class RenderModuleOperationOptions extends TargetedOperationOptions {
  @Required()
  @Description('Id corresponding to loaded module that should be rendered.')
  moduleId: string

  @Required()
  @Description('Name of the nestable in which the module should be rendered.')
  targetNestable: string
}
