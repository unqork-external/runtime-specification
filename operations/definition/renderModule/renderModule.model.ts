import { DiscriminatorValue, Required } from '@tsed/schema'

import { RenderModuleOperationOptions } from './renderModule.options'
import { Alpha } from '../../../../decorators'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.RENDER_MODULE)
@Alpha()
export class RenderModuleOperation extends Operation<OperationTypes.RENDER_MODULE> {
  public type = OperationTypes.RENDER_MODULE as const

  @Required()
  public options: RenderModuleOperationOptions = new RenderModuleOperationOptions()
}
