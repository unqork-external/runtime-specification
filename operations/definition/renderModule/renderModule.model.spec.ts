import { RenderModuleOperation } from './renderModule.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: RenderModule Tests', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(RenderModuleOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
