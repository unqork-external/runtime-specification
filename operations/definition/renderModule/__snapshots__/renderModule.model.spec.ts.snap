// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: RenderModule Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "InputRef": {
      "description": "Inputs can be used to utlize dynamic values from other state within the runtime
  within a particular execution of an operation",
      "properties": {
        "alias": {
          "description": "Optionally alias resolved value for use in formulas",
          "type": "string",
        },
        "propertyPath": {
          "description": "Path to property on target to resolve",
          "type": "string",
        },
        "required": {
          "description": "If true, operation won't execute if targeted property has no value",
          "type": "boolean",
        },
        "silent": {
          "description": "Not used in later formulas, only used to determine if operation should execute",
          "type": "boolean",
        },
        "targetKey": {
          "description": "Key of target this input depends on",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
    "RenderModuleOperationOptions": {
      "properties": {
        "inputs": {
          "description": "
    Inputs to be used within formulas in other operation options
  ",
          "items": {
            "$ref": "#/definitions/InputRef",
          },
          "type": "array",
        },
        "moduleId": {
          "description": "Id corresponding to loaded module that should be rendered.",
          "minLength": 1,
          "type": "string",
        },
        "shouldExecute": {
          "description": "
    Conditionally execute this operation based on a formula.
    Can set to false to temporarily disabled when debugging
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "Key or path that this operation should target or operate upon",
          "examples": [
            "myComponent",
            "panel.*",
            "grid.row(#).col(name)",
          ],
          "minLength": 1,
          "type": "string",
        },
        "targetNestable": {
          "description": "Name of the nestable in which the module should be rendered.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "moduleId",
        "targetNestable",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/RenderModuleOperationOptions",
    },
    "type": {
      "const": "RENDER_MODULE",
      "examples": [
        "RENDER_MODULE",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
